lib/Conversion/ExportVerilog/PrepareForEmission.cpp

```diff
@@ -115,22 +108,28 @@ static void lowerInstanceResults(InstanceOp op) {
     auto result = op.getResult(nextResultNo);
     ++nextResultNo;
 
-    if (onlyUseIsAssign(result))
-      continue;
-
+    Value newWire;
     if (result.hasOneUse()) {
       OpOperand &use = *result.getUses().begin();
       if (dyn_cast_or_null<OutputOp>(use.getOwner()))
         continue;
+      if (auto assign = dyn_cast_or_null<AssignOp>(use.getOwner())) {
+        newWire = assign.dest();
+        // NOTE: Erase the assign op to simplify the implementation. We will
+        // re-create the same assignment at the end.
+        assign.erase();
```

Comment:
This causes problems because there is later walking of the `*result.getUses()` which will incorrectly try to walk the erased assign. 😬  (I'm seeing intermittent failures on top-of-tree.)

Reply:
Thanks. Fixed by https://github.com/llvm/circt/commit/fa6fda2217bfe1611d75b85b4e53d21a325ab09f.